block-producer/src/batch_builder/mod.rs

```diff
@@ -0,0 +1,78 @@
+use std::{sync::Arc, time::Duration};
+
+use async_trait::async_trait;
+use miden_objects::transaction::ProvenTransaction;
+use tokio::{sync::RwLock, time};
+
+use crate::txqueue::TransactionQueueTrait;
+
+pub struct TransactionBatch {
+    batch: Vec<Arc<ProvenTransaction>>,
+}
```

Comment:
Probably not for this PR, but we should differentiate between "proposed" and "proven" transaction batches:

- Proposed transaction batch is just a list of transaction - same as this struct.
- Proven transaction batch would not have any transaction but would contain summary information about the cumulative effects of the batch on the state. For example, a list of account states to update, a list of new notes to add to the states, a list of new nullifiers to add to the state.